celery/backends/mongodb.py

```diff
@@ -269,14 +269,6 @@ def __reduce__(self, args=(), kwargs=None):
     def _get_database(self):
         conn = self._get_connection()
         db = conn[self.database_name]
-        if self.user and self.password:
```

Comment:
Don't we need to check the client's version first or are we dropping support for py-mongo<3.5?